lazyflow/request/threadPool.py

```diff
@@ -87,98 +101,44 @@ class _Worker(threading.Thread):
     def __init__(self, thread_pool, index):
         super().__init__(name=f"Worker #{index}", daemon=True)
         self.thread_pool = thread_pool
-        self.stopped = False
-        self.job_queue_condition = threading.Condition()
-        self.job_queue = queue.PriorityQueue()
+        self.job_queue = queue.Queue()
         self.state = "initialized"
 
+    def get_next_task(self):
+        self.thread_pool.ready_workers.put(self)
+        return self.job_queue.get()
+
     def run(self):
         """Keep executing available tasks until we're stopped."""
         # Try to get some work.
         self.state = "waiting"
-        next_task = self._get_next_job()
 
-        while not self.stopped:
-            # Start (or resume) the work by switching to its greenlet
+        logger.info("Started worker %s", self)
+
+        while True:
+            next_task = self.get_next_task()
```

Comment:
Should we keep track of state here, too?

```suggestion
            self.state = "waiting"
            next_task = self.get_next_task()
```

Reply:
Moved `self.state = "waiting"`